README.md

```diff
@@ -1,108 +1,152 @@
+- The Prover component then invokes the Stark component to produce a proof for the committed polynomials from the Executor's state machines.
+- When tasked by the Aggregator service to produce an aggregated proof:
+  - The Prover component amalgamates the results of two previously computed batch or aggregated proofs, supplied by the Aggregator, to create an aggregated proof.
+- When tasked by the Aggregator service to produce a final proof:
+  - The Prover component uses the outcome of a prior aggregated proof, supplied by the Aggregator, to formulate a conclusive proof that can be validated.
+- The server interface for this service is delineated in the file named `aggregator.proto`.
 
 ### Executor service
-- It calls the Executor component that executes the input data (a batch of EVM transactions) and calculates the resulting state.  The proof is not generated.
-- It provides a fast way to check if the proposed batch of transactions is properly built and it fits the amount of work that can be proven in one single batch.
-- When called by the Executor service, the Executor component only uses the Main state machine, since the committed polynomials are not required as the proof will not be generated.
-- The interface of this service is defined by the file executor.proto.
+
+- The Executor component processes the input data, which comprises a batch of EVM transactions, and computes the resulting state. Notably, no proof is produced.
+- This service offers a swift method to verify whether a proposed batch of transactions is correctly constructed and if it aligns with the workload that can be proven in a single batch.
+- When the Executor service invokes the Executor component, only the Main state machine is utilized. This is because the committed polynomials aren't needed, given that a proof isn't generated.
+- The service's interface is outlined in the `executor.proto` file.
 
 ### StateDB service
-- It provides an interface to access the state of the system (a Merkle tree) and the database where the state is stored.
-- It is used by the executor and the prover, as the single source of state.  It can be used to get state details, e.g. account balances.
-- The interface of this service is defined by the file statedb.proto.
 
-## Setup
+- This service provides an interface to access the system's state (represented as a Merkle tree) and the database where this state is stored.
+- Both the executor and the prover rely on it as the unified source of state. It can be utilized to retrieve specific state details, such as account balances.
+- The interface for this service is described in the `statedb.proto` file.
+
+## Compiling locally
 
+Steps to compile `zkevm-prover` locally:
 ### Clone repository
+
 ```sh
-$ git clone git@github.com:0xPolygonHermez/zkevm-prover.git
-$ cd zkevm-prover
-$ git submodule init
-$ git submodule update
+git clone git@github.com:0xPolygonHermez/zkevm-prover.git
+cd zkevm-prover
+git submodule init
+git submodule update
```

Comment:
we can remove this two lines:
```
git submodule init
git submodule update
```
if we add the flag `--recursive ` to the clone command:
```
git clone --recursive https://github.com/0xPolygonHermez/zkevm-prover.git
```